lib/src/core/authentication-core.ts

```diff
@@ -438,75 +438,49 @@ export class AuthenticationCore<T> {
             return Promise.resolve();
         }
 
-        if ((configData as any).wellKnownEndpoint) {
-            const wellKnownEndpoint = (configData as any).wellKnownEndpoint;
+        const wellKnownEndpoint = await this._authenticationHelper.resolveWellKnownEndpoint();
 
-            let response: Response;
+        let response: Response;
 
-            try {
-                response = await fetch(wellKnownEndpoint);
-                if (response.status !== 200 || !response.ok) {
-                    throw new Error();
-                }
-            } catch {
-                throw new AsgardeoAuthException(
-                    "JS-AUTH_CORE-GOPMD-HE01",
-                    "Invalid well-known response",
-                    "The well known endpoint response has been failed with an error."
-                );
+        try {
+            if (!configData?.sendWellKnownEndpointRequest) {
+                throw new Error();
             }
 
-            await this._dataLayer.setOIDCProviderMetaData(
-                await this._authenticationHelper.resolveEndpoints(await response.json())
-            );
-            await this._dataLayer.setTemporaryDataParameter(OP_CONFIG_INITIATED, true);
-
-            return Promise.resolve();
-        } else if ((configData as any).baseUrl || (configData as any).serverOrigin) {
-            try {
-                await this._dataLayer.setOIDCProviderMetaData(
-                    await this._authenticationHelper.resolveEndpointsByBaseURL());
-            } catch (error: any) {
-                throw new AsgardeoAuthException(
-                    "JS-AUTH_CORE-GOPMD-IV02",
-                    "Resolving endpoints failed.",
-                    error ?? "Resolving endpoints by base url failed."
-                );
+            response = await fetch(wellKnownEndpoint);
+            if (response.status !== 200 || !response.ok) {
+                throw new Error();
             }
+        } catch {
+            await this._dataLayer.setOIDCProviderMetaData(await this._authenticationHelper.resolveFallbackEndpoints());
```

Comment:
We are not going to use fallbacks, right?

Reply:
Yes will work on this